src/traffic_cop/traffic_cop_util.cpp

```diff
@@ -51,11 +53,27 @@ std::unique_ptr<planner::AbstractPlanNode> TrafficCopUtil::Optimize(
       auto types = order_by->GetOrderByTypes();
       auto exprs = order_by->GetOrderByExpressions();
       for (size_t idx = 0; idx < order_by->GetOrderByExpressionsSize(); idx++) {
+        if (exprs[idx].Get()->GetExpressionType() == terrier::parser::ExpressionType::VALUE_CONSTANT) {
```

Comment:
Hi! Thanks for the PR.

You might want to consider moving the logic into the Binder instead, see [0]. I think you might be able to do this as part of the Binder's visitor pattern, see [1] for an example (essentially, you are given that [2] will be called, which will call [3], and your code can piggyback on that).

Also, please note that whenever you create a new expression (the parser::ColumnValueExpression below) that you should figure out who's going to own it. If you're in the Binder, you could push it into the sherpa's ParseResult, e.g. [4].

1. This would let you support both ORDER BY (number) and ORDER BY (column name) with pretty much the same code, e.g., you could turn the column name into the index and use your existing code.
2. This lets you throw errors for "invalid column index" and "invalid table name" -- these aren't parsing errors, so the earliest place you can throw it is in the binder.


(all psql snippets from postgres)

```
psql=> select * from foo order by bar;
ERROR:  column "bar" does not exist
LINE 1: select * from foo order by bar;
```

```
psql=> select * from foo order by 2;
ERROR:  ORDER BY position 2 is not in select list
LINE 1: select * from foo order by 2;
```

[0] https://github.com/cmu-db/terrier/blob/master/src/binder/bind_node_visitor.cpp#L728
[1] https://github.com/cmu-db/terrier/pull/831/files#diff-30e70da0e63e3710075e791cb17a2caaR56
[2] https://github.com/cmu-db/terrier/blob/master/src/binder/bind_node_visitor.cpp#L466
[3] https://github.com/cmu-db/terrier/blob/master/src/binder/bind_node_visitor.cpp#L728
[4] https://github.com/cmu-db/terrier/blob/master/src/binder/bind_node_visitor.cpp#L384

Reply:
Thank you very much for your suggestion.